ach/src/main/java/com/adyen/checkout/ach/DefaultACHDirectDebitDelegate.kt

```diff
@@ -0,0 +1,357 @@
+                val countryOptions = AddressFormUtils.initializeCountryOptions(
+                    shopperLocale = componentParams.shopperLocale,
+                    addressParams = componentParams.addressParams,
+                    countryList = countries
+                )
+                countryOptions.firstOrNull { it.selected }?.let {
+                    inputData.address.country = it.code
+                    requestStateList(it.code)
+                }
+                updateOutputData(countryOptions = countryOptions)
+            }
+            .launchIn(coroutineScope)
+    }
+
+    private fun subscribeToStatesList() {
+        addressRepository.statesFlow
+            .distinctUntilChanged()
+            .onEach { states ->
+                Logger.d(TAG, "New states emitted - states: ${states.size}")
+                updateOutputData(stateOptions = AddressFormUtils.initializeStateOptions(states))
+            }
+            .launchIn(coroutineScope)
+    }
+
+    private fun updateOutputData(
+        countryOptions: List<AddressListItem> = outputData.addressState.countryOptions,
+        stateOptions: List<AddressListItem> = outputData.addressState.stateOptions,
+    ) {
+        val newOutputData = createOutputData(countryOptions, stateOptions)
+        _outputDataFlow.tryEmit(newOutputData)
+        updateComponentState(newOutputData)
+    }
+
+    private fun requestStateList(countryCode: String?) {
+        addressRepository.getStateList(
+            shopperLocale = componentParams.shopperLocale,
+            countryCode = countryCode,
+            coroutineScope = coroutineScope
+        )
+    }
+
+    private fun requestCountryList() {
+        addressRepository.getCountryList(
+            shopperLocale = componentParams.shopperLocale,
+            coroutineScope = coroutineScope
+        )
+    }
+
+    private fun sendAnalyticsEvent(coroutineScope: CoroutineScope) {
+        Logger.v(TAG, "sendAnalyticsEvent")
+        coroutineScope.launch {
+            analyticsRepository.sendAnalyticsEvent()
+        }
+    }
+
+    private fun updateComponentState(outputData: ACHDirectDebitOutputData) {
+        Logger.v(TAG, "updateComponentState")
+        val componentState = createComponentState(outputData)
+        _componentStateFlow.tryEmit(componentState)
+    }
+
+    @Suppress("ReturnCount")
+    private fun createComponentState(
+        outputData: ACHDirectDebitOutputData = this.outputData
+    ): PaymentComponentState<ACHDirectDebitPaymentMethod> {
+        val publicKey = publicKey
+        if (!outputData.isValid || publicKey == null) {
+            return PaymentComponentState(
+                data = PaymentComponentData(),
+                isInputValid = outputData.isValid,
+                isReady = publicKey != null
+            )
+        }
+
+        try {
+            val encryptedBankAccountNumber = genericEncrypter.encryptField(
+                encryptionKey = BANK_ACCOUNT_NUMBER,
```

Comment:
Can you also move `GenericEncrypter.KCP_PASSWORD_KEY` to the companion of `DefaultCardDelegate`? It's not exactly in the scope of this PR but we discussed it with this task previously

Reply:
Can we also call is `ENCRYPTION_KEY_FOR_ KCP_PASSWORD` just like the other 2?